src/chess/position.cc

```diff
@@ -137,4 +142,87 @@ uint64_t PositionHistory::HashLast(int positions) const {
   return HashCat(hash, Last().GetNoCaptureNoPawnPly());
 }
 
+// PrintFen outputs a FEN notation of the board. 
+// based on https://en.wikipedia.org/wiki/Forsyth%E2%80%93Edwards_Notation#Examples
+string Position::GetFen() const  {
+	string result;
+	string enpassant;
+	ChessBoard board = GetBoard();
+	bool fenflipped = false;
+	if (board.flipped()) {
+		board.Mirror();
+		fenflipped = true;
+	}
+	int emptycounter = 0;
+	for (int i = 7; i >= 0; --i) {
+		for (int j = 0; j < 8; ++j) {
+			if (emptycounter > 0 &&
+				(
+				    board.our_king().get(i,j) ||
```

Comment:
I think this logic should be moved to another function. Other a free-standing one or member function of `ChessBoard` class. Something like:

```cpp
char GetPieceAt(const ChessBoard& board, int row, int col) {
  BoardSquare sq(row, col);
  if (!(ours() | theirs()).get(sq)) return '\0';
  if (board.our_king() == sq) return 'K';
  if (board.their_king() == sq) return 'k';
  char res = 'N';
  if (board.bishops().get(sq)) res = 'B';
  else if (board.queens().get(sq)) res = 'Q';
  else if (board.rooks().get(sq)) res = 'R';
  return flipped() ? std::tolower(res) : res;
}